mm2src/crypto/src/mnemonic.rs

```diff
@@ -160,26 +154,24 @@ mod tests {
         let decrypted_mnemonic = decrypted_mnemonic.unwrap();
 
         // Verify if decrypted mnemonic matches the original
-        assert_eq!(decrypted_mnemonic, parsed_mnemonic);
+        assert_eq!(decrypted_mnemonic, mnemonic);
     });
 
-    cross_test!(test_mnemonic_with_last_byte_zero, {
-        let mnemonic = "tank abandon bind salon remove wisdom net size aspect direct source fossil\0".to_string();
-        let password = "password";
+    cross_test!(test_encrypt_decrypt_words, {
+        let mnemonic = "Helloworld";
+        let password = "Helloworld";
 
         // Encrypt the mnemonic
-        let encrypted_data = encrypt_mnemonic(&mnemonic, password);
+        let encrypted_data = encrypt_mnemonic(mnemonic, password);
         assert!(encrypted_data.is_ok());
         let encrypted_data = encrypted_data.unwrap();
 
         // Decrypt the mnemonic
         let decrypted_mnemonic = decrypt_mnemonic(&encrypted_data, password);
-        assert!(decrypted_mnemonic.is_err());
+        assert!(decrypted_mnemonic.is_ok());
+        let decrypted_mnemonic = decrypted_mnemonic.unwrap();
 
-        // Verify that the error is due to parsing and not padding
-        assert!(decrypted_mnemonic
-            .unwrap_err()
-            .to_string()
-            .contains("mnemonic contains an unknown word (word 11)"));
```

Comment:
I would like to keep this test as it was as it covers the case of making sure padding works fine. You can use `Mnemonic::parse_normalized` to make it work as before.

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2312/commits/488b91982a209f56d1a934706642814adfd71dcd